functions-continued/variable-scope.md

```diff
@@ -0,0 +1,396 @@
+2. Predict what the code will print out
+3. Run the code
+4. Compare your predictions against what you actually see
+5. Write down your conclusions about how local variables "leave" at the end of each function call, and get re-initialized at the beginning of each function call.
+
+```python
+def display_breakfast(breakfast_of_champions):
+    # The parameter breakfast_of_champions exists in this function!
+    print(f"This function call has the parameter breakfast_of_champions with the value {breakfast_of_champions}")
+
+    # The local variable breakfast_message exists in this function!
+    breakfast_message = f"My breakfast today is {breakfast_of_champions}"
+    print(f"{breakfast_message}")
+
+    # After the return statement, the function will end, and breakfast_of_champions AND breakfast_message will no longer exist...
+    return breakfast_message
+
+display_breakfast("waffles and syrup")
+# print(f"Uncommenting this line and trying to access breakfast_of_champions will create a NameError: {breakfast_of_champions}")
+# print(f"Uncommenting this line and trying to access breakfast_message will create a NameError: {breakfast_message}")
+display_breakfast("rice and soup")
+display_breakfast("coffee and toast")
+# print(f"Uncommenting this line and trying to access breakfast_of_champions will create a NameError: {breakfast_of_champions}")
+# print(f"Uncommenting this line and trying to access breakfast_message will create a NameError: {breakfast_message}")
+```
+
+At first, it may be frustrating that we can't access `breakfast_of_champions` or `breakfast_message` outside of the function.
+
+On the bright side, it makes our code extremely consistent and predictable: Python will **never** think `breakfast_of_champions` has any value, except when we pass a value into the `display_breakfast` function call. Programming and using variables to know what values they **aren't** is just as valuable as knowing what values they are.
+
+### Our Strategies Will Evolve!
+
+As we learn about more concepts, such as _iteration_ and _classes_, we'll see that local variables have special scope within those kinds of blocks.
+
+It will be crucial to understand the local variable scope for those cases, too, and to rely on them.
+
+Looking ahead: When we learn about _iteration_ in programming, most syntaxes for iteration create their own variable scope.
+
+In this example, follow these steps in order:
+
+1. Read this code and make a prediction about what it does and what will happen when you run it
+2. Run it
+3. Make predictions about where the local variables are, and what rules about scope they may follow
+4. Write down your predictions
+
+```python
+def announce_lunch_menu(lunch_items):
+    print("It is now time to declare each lunch item, slowly, and dramatically.")
+    for lunch_item in lunch_items:
+        print("Today we will eat......")
+        print("a.....")
+        print(lunch_item)
+    print("I hope you're hungry!")
```

Comment:
I'm not sure of the value of including this sample here. If iteration is covered elsewhere, and there's nothing special about scope as it relates to the surrounding discussion, this just seems to muddy the water.

I would consider cutting the entire 'Our Strategies Will Evolve!' section so that iteration gets the full discussion it needs rather than possible confusing things by squeezing this aside in here.